src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -188,8 +188,12 @@ public BucketMetadata getBucketMetadata(long id) {
 
     public List<BucketMetadata> listBuckets(String ownerName, String kind) {
         List<BucketEntity> entities;
+        List<String> owners = new ArrayList<>();
+        owners.add(ownerName);
 
-        if (!StringUtils.isEmpty(ownerName)) {
+        if (!StringUtils.isEmpty(ownerName) && !StringUtils.isEmpty(kind)) {
+            entities = bucketRepository.findByOwnerIsInContainingKind(owners, kind);
```

Comment:
You can use ```Collections.singletonList(ownerName)``` instead of:  
  ```     
List<String> owners = new ArrayList<>();
       owners.add(ownerName);
```